### PR TITLE
[collector] Add option to set the nodePort mapping

### PR DESCRIPTION
**Pull request by mllab-nl**
Fixes https://github.com/open-telemetry/opentelemetry-helm-charts/issues/380

### CODE REVIEW

**Review by TylerHelmuth — COMMENTED** [review]
@mllab-nl thanks for your contribution.  Modifying the schema.json will be necessary for resolving the linked issue, but you'll also need to update the values.yaml and templates.

Once everything is implemented, please bump the patch version of the chart and then run `make generate-examples CHARTS=opentelemetry-collector`

**Comment by mllab-nl** [comment]
@TylerHelmuth nodePort is only make sense when you use NodePort service type and only when you specifically want to. 
What changes would you expect to values.yaml and templates? 

I will bump up the patch version

**Comment by TylerHelmuth** [comment]
@mllab-nl I believe for the new `nodePort` field to be used you'll need to update https://github.com/open-telemetry/opentelemetry-helm-charts/blob/808ed313d4283f2bfe123eea12cb0a67691e21a9/charts/opentelemetry-collector/templates/_config.tpl#L295 to set the `nodePort`.  That template is used by service.yaml to set the service's ports.

Lets also add a comment in the values.yaml's port section that specifies that `nodePort` is an allowed field.

**Comment by mllab-nl** [comment]
Thx. For guidance. Done. 
Bumped the version again. 

What is your approach to testing and examples - what is their primary role examples or tests  ?   
Did a test in charts\opentelemetry-collector\examples\deployment-otlp-traces\
Current config of the example renders the ClusrterIp service with Node port which is expected, but service is not functional - ClusterIP service with a NodePort.
IMO adding an additional example for NodePort only is a bit too much, since all those are always regenerated and checked in. 

Should the test stay or should I revert it ?

File: charts/opentelemetry-collector/examples/deployment-otlp-traces/values.yaml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 mode: deployment
 
 ports:
+  otlp:
+    enabled: true
+    nodePort: 30317
```
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/examples/deployment-otlp-traces/values.yaml:6]: @mllab-nl we use the rendered examples to help users with common situations or complicated configuration.  I don't think `nodePort` is going to be common nor is it a complicated field to use when needed so I don't think we need to update any examples.

For your local testing you should create a values.yaml with all the fields set that will test `nodePort` (you can change service type with `service.type` in the values.yaml)

**Comment by Allex1** [comment]
@mllab-nl pls use a meaningful title/commit name

File: charts/opentelemetry-collector/examples/deployment-otlp-traces/values.yaml (hunk 1 above)
Comment (mllab-nl) [reply charts/opentelemetry-collector/examples/deployment-otlp-traces/values.yaml]: Removed. 
Local check confirm that with following config I get proper service with correct ports deployed:

> 
> 
> mode: deployment
> 
> ports:
>   otlp:
>     nodePort: 30117
>   otlp-http:    
>     nodePort: 30118
>   jaeger-compact:
>     enabled: false
>   jaeger-thrift:
>     enabled: false
>   jaeger-grpc:
>     enabled: false
>   zipkin:
>     enabled: false
> 
> service:
>   type: NodePort

**Comment by mllab-nl** [comment]
@TylerHelmuth Updated the fork, resolved conflicts, tested  - Ok
Thank you !

**Pull request merged**